pyproject.toml

```diff
@@ -1,3 +1,103 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "pytest-runner",
+  "setuptools>=61.2",
+]
+
+[project]
+name = "kornia"
+version = "0.6.11.dev"
+description = "Open Source Differentiable Computer Vision Library for PyTorch"
+[project.readme]
+file = "README.md"
+content-type = "text/markdown"
+
+keywords = [
+  "computer vision",
+  "deep learning",
+  "pytorch",
+]
+license = {text = "Apache-2.0"}
+authors = [{name = "Edgar Riba", email = "edgar@kornia.org"}]
+requires-python = ">=3.7"
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Environment :: Console",
+    "Environment :: GPU",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Education",
+    "Intended Audience :: Information Technology",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence",
+    "Topic :: Scientific/Engineering :: Image Processing",
+    "Topic :: Software Development :: Libraries",
+]
+dependencies = [
+  "packaging",
+  "torch>=1.9.1",
+]
+[project.optional-dependencies]
+dev = [
+  "isort",
+  "kornia-rs==0.0.8",
+  "mypy[reports]",
+  "numpy",
+  "opencv-python",
+  "pre-commit>=2",
+  "pydocstyle",
+  "pytest==7.2.1",
+  "pytest-cov==4",
+  "scipy",
+]
+docs = [
+  "furo",
+  "matplotlib",
+  "opencv-python",
+  "PyYAML<6.1.0,>=5.1",
+  "sphinx>=4",
+  "sphinx-autodoc-defaultargs",
+  "sphinx-autodoc-typehints",
+  "sphinx-copybutton>=0.3",
+  "sphinx-design",
+  "sphinx-rtd-theme>0.5",
+  "sphinxcontrib-bibtex",
+  "sphinxcontrib-gtagjs",
+  "sphinxcontrib-youtube",
+  "torchvision",
+]
+x = [
+  "accelerate==0.16",
+]
+[project.urls]
+"Bug Tracker" = "https://github.com/kornia/kornia/issues"
+Documentation = "https://kornia.readthedocs.io/en/latest"
+Download = "https://github.com/kornia/kornia"
+Homepage = "https://www.kornia.org"
+"Source Code" = "https://github.com/kornia/kornia"
+
+[tool.setuptools]
```

Comment:
when building i got: _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.

Reply:
Yes. https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration
> Warning
> Support for declaring configurations not standardized by [PEP 621](https://peps.python.org/pep-0621/) (i.e. the [tool.setuptools] table), is still in beta stage and might change in future releases.

This is why losing to the validation tool would be a bad idea.

Reply:
I didn't see any warning of this hook, but ok, just kinda concern for being a new/experimental project 😅